experiments/experiments-impl/src/main/java/com/duckduckgo/experiments/impl/reinstalls/ReinstallAtbListener.kt

```diff
@@ -51,16 +61,40 @@ class ReinstallAtbListener @Inject constructor(
             } else {
                 downloadsDirectoryManager.createNewDirectory(DDG_DOWNLOADS_DIRECTORY)
             }
-            appInstallStore.returningUserChecked = true
+            reinstallSharedPrefs.get().setReturningUserChecked()
         }
     }
 
     override fun beforeAtbInitTimeoutMillis(): Long = MAX_REINSTALL_WAIT_TIME_MS
 
+    private fun SharedPreferences.isReturningUserChecked(): Boolean {
+        return getBoolean(RETURNING_USER_CHECKED_TAG, false)
+    }
+
+    private fun SharedPreferences.setReturningUserChecked() {
+        this.edit(commit = true) { putBoolean(RETURNING_USER_CHECKED_TAG, true) }
+    }
+
     companion object {
         private const val MAX_REINSTALL_WAIT_TIME_MS = 1_500L
-
-        const val REINSTALL_VARIANT = "ru"
         private const val DDG_DOWNLOADS_DIRECTORY = "DuckDuckGo"
+        private const val RETURNING_USER_CHECKED_TAG = "RETURNING_USER_CHECKED_TAG"
+    }
+}
+
+internal const val REINSTALL_VARIANT = "ru"
+
+@Retention(AnnotationRetention.BINARY)
+@Qualifier
+private annotation class ReinstallSharedPrefs
+
+@Module
+@ContributesTo(AppScope::class)
+class IgnoringBatteryOptimizationsModule {
```

Comment:
why this class name?

Reply:
because I copy paste and didn't update the name :D